longest-increasing-subsequence/yeonguchoe.cs

```diff
@@ -0,0 +1,25 @@
+public class Solution {
+    public int LengthOfLIS(int[] nums) {
+        if (nums.Length == 0) return 0;
+
+        int[] dp = new int[nums.Length];
+        for (int i = 0; i < nums.Length; i++) {
+            dp[i] = 1;
+        }
+
+        for (int i = 1; i < nums.Length; i++) {
+            for (int j = 0; j < i; j++) {
+                if (nums[j] < nums[i]) {
+                    dp[i] = Math.Max(dp[i], dp[j] + 1);
+                }
+            }
+        }
+
+        int maxLength = 1;
+        foreach (int length in dp) {
+            maxLength = Math.Max(maxLength, length);
+        }
```

Comment:
```suggestion
        int maxLength = 1;
        for (int i = 1; i < nums.Length; i++) {
            for (int j = 0; j < i; j++) {
                if (nums[j] < nums[i]) {
                    dp[i] = Math.Max(dp[i], dp[j] + 1);
                }
            }
            maxLength = Math.max(maxLength, dp[i]);
        }
```

`maxLength`를 구하는 반복문도 L10의 반복문 안에 포함시키는 방식으로 구현하는게 더 나을 것 같습니다 :)